biblat_process/biblatjournal.py

```diff
@@ -0,0 +1,108 @@
+import inspect
+from datetime import datetime
+
+from biblat_process import tesauro
+from biblat_schema.catalogs import Disciplina
+
+
+class RevistaDict:
+
+    def __init__(self, marc_dict):
+        self.marc_dict = marc_dict
+
+    def to_dict(self):
+        properties = {}
+        for name, value in \
+                inspect.getmembers(self.__class__,
+                                   lambda o: isinstance(o, property)):
+            value = getattr(self, name)
+            if not name.startswith('__') and not inspect.ismethod(value):
+                properties[name] = value
+        return properties
+
+    @property
+    def base_datos(self):
+        # TODO revisar la etiqueta
+        return self.marc_dict.get('035', [{'a': None}])[0].get('a', None)
+
+    @property
+    def titulo_revista(self):
+        return self.marc_dict.get('222', [{'a': None}])[0].get('a', None)
+
+    @property
+    def titulo_abr_revista(self):
+        # TODO revisar la etiqueta.
+        return self.marc_dict.get('245', [{'a': None}])[0].get('a', None)
+
+    @property
+    def issn(self):
+        # TODO revisar la etiqueta
+        return self.marc_dict.get('022', [{'a': None}])[0].get('a', None)
+
+    @property
+    def pais(self):
+        # TODO revisar la etiqueta
+        pais = self.marc_dict.get('008', [{'e': None}])[0].get('e', None)
+        if pais in tesauro.paises:
+             pais = tesauro.paises[pais]
+        else:
+            pais = None
+        return pais
+
+    @property
+    def disciplina(self):
+        disc = Disciplina()
+        if '698' in self.marc_dict:
+            for disciplinadoc in self.marc_dict['698']:
+                disc.meta = disciplinadoc.get('spa', None)
+                disc.nombre = disciplinadoc.get('a', None)
+        return disc
```

Comment:
El objeto de tipo Disciplina debe ser obtenido por medio de una consulta, ya que es un campo de tipo `ReferenceField`, el campo meta no debe usarse para asignar un idioma es un atributo de la clase para especificar información especifica en la base de datos.

https://github.com/dgb-sistemas/biblat-schema/blob/master/biblat_schema/models.py#L58

Leer documentación de mongoengine
http://docs.mongoengine.org/guide/querying.html

Reply:
Ok, aunque tengo una duda. ¿Que no se supone que estamos parcenado el registro MARC desde un archivo plano? Aun no ingreso información a Mongo, por lo que aun no utilizo el objeto Revista de Mongo. ¿Deberia ingresar la informacion a Mongo?

Reply:
Con respecto al campo meta, ¿con que informacion se debe de llenar? Veo que el campo 035 contiene informacion de la base de datos. ¿Es esta la informacion que se debe ingresar en el campo meta?

Reply:
Cuando me refiero a la base de datos me refiero al manejador de la base de datos, el elemento meta ya esta definido para el objeto Disciplina, por favor revisa la documentación:
http://docs.mongoengine.org/guide/defining-documents.html

Reply:
El campo 035 contine información del atribito base_datos en este caso la información se tiene en los primeros 5 caracteres: ej:
CLA01000000500
La información de la base es "CLA01"

Reply:
@arturorc ya hice los cambios:
        if '698' in self.marc_dict:
            for disciplinadoc in self.marc_dict['698']:
                disciplinadoc = disciplinadoc.get('a', None)
                if disciplinadoc in self.disciplinas_list:
                    nombre = Disciplina.objects(__raw__={'nombre':nombre}).first()
                    disc.append(nombre)
        base = self.marc_dict.get('035', [{'a': None}])[0].get('a', None)
        base = base[:5]
        disc.append(base)
        print(disc)
        return disc or None

Sin embargo, al correr las pruebas al parecer no esta recuperando el catalogo de Mongo. Esto porque no tengo ningun catalago de Disciplinas en Mongo. ¿Que debo de hacer para tener el catalogo? ¿Me puede facilitar uno? ¿Que debo de hacer para ingresarlo a Mongo?